### DISCUSSION TITLE
Mismatch between UI timeline and review items

### DISCUSSION THREAD

**Discussion by ofcedwards** [blakeblackshear/frigate#12177] [Report a Bug] [answered]
### Describe the problem you are having

The timeline in the UI is highlighted to show detections and alerts. I'm finding that clicking on a review item commonly starts playback in the middle (not literally 50%) of the highlighted segment. The review item appears to be behaving as expected in that playback is starting 4 seconds prior to the reported start time. However, going back to the start of highlighted portion of the timeline shows what I'd consider to be the full alert/detection. It would be preferable to use the same logic the timeline has for calculating start/end times for review playback/exports.

For example: I have alerts set to only include people and to require a zone encompassing my front yard/driveway. I have a car parked in a space nearest to the camera and one parked next to it that's usually not detected because too much of it is covered by the first car. When the second car pulls in the driveway and a person gets out, the review system starts playback when the person is detected, but the timeline shows the motion from the undetected car as part of the alert as well. If I go back to the start of where the timeline is highlighted, I can watch the car come into frame, pull into the driveway, and then see the person come into frame after they get out. Checking the mqtt payloads, I can see that the review system is working exactly as intended (starting playback 4 seconds before the start of the alert, which is when the person is detected), but the timeline seems to do a better job of actually determining when the alert _should_ start.

If a car is detected first and then the review changes to an alert, the clip starts with the car coming into view as expected. There just seems to be some disparity in determining the start time of the review item when it's preceded by motion only (with no object detection).

For what it's worth, I'd prefer the review playback/export logic to match what the timeline is doing, not have the timeline shading be shortened to match where the review playback currently starts.

### Steps to reproduce

1. Find or otherwise generate a review item that's immediately preceded by motion, but not object detection
2. Click on the review item in frigate to start playback
3. Pause playback immediately on start
4. Observe placement of the red line on the timeline in reference to where the red/orange shading starts (easier to view on mobile)

### Version

0.14.0-bfbacee

### Frigate config file

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### Operating system

UNRAID

### Install method

Docker CLI

### Network connection

Mixed

### Camera make and model

N/A

**Comment by NickM-27** [comment]
the times shown on the timeline and the times used for starting playback are identical, it would be impossible for the inputs to be different because the same data feeds both of those. If anything this is just an issue of the browser seeking to the wrong time.

I also don't see this behavior personally, it always starts at the beginning of the review item in the timeline 

![Screen Shot 2024-06-27 at 13 03 17 PM](https://github.com/blakeblackshear/frigate/assets/14866235/c49faab5-2edc-4c2e-b86f-4258625ead5f)

**Comment by NickM-27** [comment]
a visual example might be more helpful to understand what you are describing, in case I am misunderstanding what you mean

**Reply by ofcedwards** [reply]
Can I email a screen recording to you?

**Reply by NickM-27** [reply]
Sure, nickmowen213@gmail.com

**Reply by NickM-27** [reply]
At the current zoom level there are only 8 pixels to represent 30 seconds, getting it down to the pixel would make many parts of the timeline close to invisible. This is working as intended currently.

**Reply by ofcedwards** [reply]
Is the timeline shading locked to every :00 and :30 on each minute?

**Reply by hawkeye217** [reply, marked as answer]
Yes. If a review item begins anywhere in that segment, it is shaded. In your example above, if the review item begins at 6:45:14, the entire 6:45:00-6:45:29 segment is shaded.

**Reply by ofcedwards** [reply]
Alright, that explains what I'm seeing. I first noticed this on reviews that start almost at the end of the 30 second block.

On desktop (usually a 13" MacBook for me) this isn't so noticeable, but it's pretty apparent that the shading is off with the larger timeline on mobile. It would make sense to have the precision of the shading change with the zoom level of the timeline. Should I add this suggestion to the feature request that's open for timeline zooming or open a new feature request for it?

**Reply by NickM-27** [reply]
it won't change at this zoom level, the request in general for zooming will _inherently_ adjust this when zoomed in more. Feel free to leave a comment in that feature request
